rstcheck/__init__.py

```diff
@@ -516,18 +525,48 @@ def load_configuration_from_file(directory: str, args: argparse.Namespace) -> ar
 
 
 def _get_options(directory_or_file: str, debug: bool = False) -> typing.Dict[str, str]:
+    options = {}
+
     config_path = find_config(directory_or_file, debug=debug)
-    if not config_path:
-        return {}
+    if config_path:
+        config_file = os.path.basename(config_path)
 
-    parser = configparser.ConfigParser()
-    parser.read(config_path)
-    try:
-        options = dict(parser.items("rstcheck"))
-    except configparser.NoSectionError:
-        return {}
+    if config_file == "pyproject.toml":
+        options = _get_pyproject_options(config_path)
     else:
-        return options
+        parser = configparser.ConfigParser()
+        parser.read(config_path)
+        try:
+            options = dict(parser.items("rstcheck"))
+        except configparser.NoSectionError:
+            options = {}
+
+    return options
+
+
+def _get_pyproject_options(config_path: str) -> typing.Dict[str, str]:
+    with open(config_path, "rb") as conf_file:
+        config = tomli.load(conf_file)
+        options: typing.Dict[str, str] = config.get("tool", {}).get("rstcheck", {})
```

Comment:
Mypy cannot control this, but the type hint is missing the `List[str]` doesn't it? Because the ignore_* configs are actually lists.

But see other comment before changing anything.

Reply:
This is a must change for me.